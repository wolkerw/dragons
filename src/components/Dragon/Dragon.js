import React from "react";
import { Row } from "react-bootstrap";
import PropTypes from "prop-types";
import _ from "lodash";
import ButtonStandard from "../../components/ButtonStandard/ButtonStandard";
import "./Dragon.scss";

function Dragon(props) {
    const { data, removeDragon, editDragon } = props;

    const lodashTruncate = (data, maxLength) => {
        return _.truncate(data, {
            length: maxLength
        });
    };

    if (data !== undefined && typeof data == "object") {
        const newData = {
            name: data.name,
            type: data.type
        };

        let cols = 12;
        const objects = Object.entries(newData);

        const math = Number((cols / objects.length).toFixed(1));

        const size = math > 1 ? math : 1;

        const maxLength = size * 12;

        const colRemove = (
            <div className="col-2 custom-col">
                <div className="custom-checkbox">
                    <ButtonStandard
                        text="Remover"
                        textClass="text-2"
                        className="btn btn-primary btn-rounded btn-standard px-4 ml-3"
                        onClick={() => removeDragon(data.id)}
                    />
                </div>
            </div>
        );

        const colEdit = (
            <div className="col-2 custom-col">
                <div className="custom-checkbox">
                    <ButtonStandard
                        text="Editar"
                        textClass="text-2"
                        className="btn btn-primary btn-rounded btn-standard px-4 ml-3"
                        onClick={() => editDragon(data.id)}
                    />
                </div>
            </div>
        );

        if (objects.length > 0) {
            return (
                <Row className="custom-dragon-row my-2">
                    {objects.map((data, i) => {
                        return (
                            <div className="col-4" key={i}>
                                <span className="text-2 font-weight-bolder">
                                    {lodashTruncate(data[1], maxLength)}
                                </span>
                            </div>
                        );
                    })}
                    {colRemove}
                    {colEdit}
                </Row>
            );
        } else {
            return <p>Nenhum dragão encontrado.</p>;
        }
    } else {
        return <p>Dados inválidos</p>;
    }
}

Dragon.propTypes = {
    data: PropTypes.object.isRequired,
    id: PropTypes.string
};
export default Dragon;
