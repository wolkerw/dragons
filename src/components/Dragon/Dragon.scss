@import "../../styles/base/base";

//Row
.custom-dragon-row {
    color: $text-color-primary;
    padding: 10px 0 10px 0;
    background-color: $color-secondary;
    margin: auto !important;

    &:hover {
        background-color: $color-secondary-hover;
        cursor: pointer;
    }

    [class^="col-"] {
        white-space: nowrap;
        overflow: hidden;
    }

    span {
        font-size: 0.8em !important;
        font-weight: bolder !important;
    }
}

.custom-col {
    overflow: visible !important;
}

.custom-col-status {
    [class*="status"] {
        margin-top: 3px;
    }
}