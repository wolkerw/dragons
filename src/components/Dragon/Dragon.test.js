import React from "react";
import Dragon from "./Dragon";
import { IntlProvider } from "react-intl";
import { configure, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

const country = "pt-BR";

const dragonData = {
    id: "5",
    name: "Viserion",
    type: "Ã”mega"
};

describe("Dragon Component", () => {
    const component = (
        <IntlProvider locale={country}>
            <Dragon
                data={dragonData}
                functionProps={() => true}
                id={dragonData.id}
            />
        </IntlProvider>
    );

    it("matches the snapshot", () => {
        const tree = mount(component);
        expect(tree).toMatchSnapshot();
    });

    it("generated a dragon item with the edit button", () => {
        const wrapper = mount(component);
        const checkbox = wrapper.find("button");
        expect(checkbox.length).toEqual(2);
    });

    it("generated a dragon item with the columns showing", () => {
        const wrapper = mount(component);
        const checkbox = wrapper
            .find(".text-2")
            .first()
            .text();
        expect(checkbox).toBe("Viserion");
    });
});
