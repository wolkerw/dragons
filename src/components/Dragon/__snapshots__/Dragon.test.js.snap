// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dragon Component matches the snapshot 1`] = `ReactWrapper {}`;
